test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -500,32 +473,25 @@ func runtimeHookTestHandler(ctx context.Context, c client.Client, namespace, clu
 	Eventually(func() bool {
 		return blockingCondition()
 	}, intervals...).Should(BeFalse(),
-		fmt.Sprintf("ClusterTopology reconcile did not unblock after updating hook response: hook %s still blocking", hookName))
+		fmt.Sprintf("ClusterTopology reconcile did proceed as expected when calling %s", hookName))
 }
 
-// clusterConditionShowsHookFailed checks if the TopologyReconciled condition message contains both the hook name and hookFailedMessage.
-func clusterConditionShowsHookFailed(cluster *clusterv1.Cluster, hookName string) bool {
-	return conditions.GetReason(cluster, clusterv1.TopologyReconciledCondition) == clusterv1.TopologyReconcileFailedReason &&
-		strings.Contains(conditions.GetMessage(cluster, clusterv1.TopologyReconciledCondition), hookFailedMessage) &&
+// clusterConditionShowsHookBlocking checks if the TopologyReconciled condition message contains both the hook name and hookFailedMessage.
+func clusterConditionShowsHookBlocking(cluster *clusterv1.Cluster, hookName string) bool {
+	return conditions.GetReason(cluster, clusterv1.TopologyReconciledCondition) == clusterv1.TopologyReconciledHookBlockingReason &&
 		strings.Contains(conditions.GetMessage(cluster, clusterv1.TopologyReconciledCondition), hookName)
 }
 
 func dumpAndDeleteCluster(ctx context.Context, proxy framework.ClusterProxy, namespace, clusterName, artifactFolder string) {
 	By("Deleting the workload cluster")
-	cluster := &clusterv1.Cluster{}
-	Eventually(func() error {
-		return proxy.GetClient().Get(ctx, client.ObjectKey{Namespace: namespace, Name: clusterName}, cluster)
-	}).Should(Succeed())
+
+	cluster := framework.GetClusterByName(ctx, framework.GetClusterByNameInput{Name: clusterName, Namespace: namespace, Getter: proxy.GetClient()})
 
 	// Dump all the logs from the workload cluster before deleting them.
 	proxy.CollectWorkloadClusterLogs(ctx, cluster.Namespace, cluster.Name, filepath.Join(artifactFolder, "clusters-beforeClusterDelete", cluster.Name))
 
 	// Dump all Cluster API related resources to artifacts before deleting them.
-	framework.DumpAllResources(ctx, framework.DumpAllResourcesInput{
-		Lister:    proxy.GetClient(),
-		Namespace: namespace,
-		LogPath:   filepath.Join(artifactFolder, "clusters-beforeClusterDelete", proxy.GetName(), "resources"),
-	})
+	framework.DumpAllResources(ctx, framework.DumpAllResourcesInput{Lister: proxy.GetClient(), Namespace: namespace, LogPath: filepath.Join(artifactFolder, "clusters-beforeClusterDelete", proxy.GetName(), "resources")})
```

Comment:
nit: Those statements were a little bit easier to read before (of course it works on big screens...)

Reply:
Yeah - it's just a lot of vertical space for repetitive key values with most of these input structs. I'm fine either way though.

Reply:
Let's say there are folks which have a hard-limit somewhere between 100 and 120 character's. 
So I personally try not to go over ~140 or something :P 

Given that we consistently do it multi-line across the entire e2e tests / test framework, I personally would use multi-line here as well.

I won't block the PR because of that though

Reply:
(disclaimer, edited the comment multiple times)